src/uct/tcp/tcp_ep.c

```diff
@@ -1292,7 +1292,7 @@ ucs_status_t uct_tcp_ep_am_short(uct_ep_h uct_ep, uint8_t am_id, uint64_t header
     uct_tcp_iface_t *iface = ucs_derived_of(uct_ep->iface, uct_tcp_iface_t);
     uct_tcp_am_hdr_t *hdr  = NULL;
     struct iovec iov[UCT_TCP_EP_AM_SHORTV_IOV_COUNT];
-    uint32_t payload_length;
+    uint32_t UCS_V_UNUSED payload_length;
```

Comment:
looks like it is not needed at all, can use `hdr->length` instead.
minor: also, can have common UCT_TL_EP_STAT_OP(&ep->super, AM, SHORT, payload_length) invocation, outside the `if/else` (since this code will need to be updated anyway)

Reply:
`hdr->length` couldn't be used since memory pointed to by `hdr` pointer could be destroyed in `uct_tcp_ep_am_send()`

Reply:
ok

Reply:
I understand that is not clear code, I guess having a comment for `payload_length` would be great